app.rb

```diff
@@ -0,0 +1,128 @@
+# app.rb
+
+require_relative 'library'
+
+# General method to print a title
+def print_title(title)
+  puts "\n_____ #{title} _____\n\n"
+end
+
+# This class is responsible for the user interface and for
+# calling the methods of the Library class.
+class App
+  def initialize
+    @library = Library.new
+  end
+
+  def list_books(title = 'List of Books')
+    print_title('List of Books') unless title.empty?
+    books = @library.list_books
+    books.each.with_index { |book, index| puts "#{index + 1} -> #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_people(title = 'List of People')
+    print_title(title) unless title.empty?
+    people = @library.list_people
+    people.each.with_index { |person, index| puts "#{index + 1} -> #{person.name} (#{person.class.name})" }
+  end
+
+  def create_person
+    print_title('Create a Person')
+    type = person_type
+    name = person_name
+    age = person_age
+    save_person(type, name, age)
+  end
+
+  def save_person(type, name, age)
+    case type
+    when 'T'
+      create_teacher(name, age)
+    when 'S'
+      permission = parent_permission
+      create_student(name, age, permission)
+    else
+      puts "\n Invalid person type."
+    end
+  end
+
+  def person_type
+    print 'Enter person type (T -> teacher or S -> student): '
+    gets.chomp.upcase
+  end
+
+  def person_name
+    print 'Enter person name: '
+    gets.chomp.upcase
+  end
+
+  def person_age
+    print 'Enter person age: '
+    gets.chomp.to_i
+  end
+
+  def parent_permission
+    print 'Does the student have parent permission? (Y / N): '
+    gets.chomp.upcase == 'Y'
+  end
+
+  def create_teacher(name, age)
+    @library.create_teacher(name, age)
+  end
```

Comment:
- [ ] Please make sure that when you are creating a teacher, you ask for the specialization of the teacher. :pray:

Reply:
✔️ Changes are done! Thank you 👍🏽